agent/consul/leader_connect_test.go

```diff
@@ -253,7 +260,7 @@ func TestLeader_SecondaryCA_IntermediateRenew(t *testing.T) {
 	// however, defaultQueryTime will be configurable and we con lower it
 	// so that it returns for sure.
 	retry.Run(t, func(r *retry.R) {
-		secondaryProvider, _ := s2.getCAProvider()
+		secondaryProvider, _ = getCAProvider(s2)
```

Comment:
reuse the variable from above instead of creating a new one in the retry block so that the updated secondary provider is available further down to generate the leaf cert. We think this is the main issue with this test.